src/Access/UserPolicy.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Nicknames\Access;
+
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\User\Access\AbstractPolicy;
+use Flarum\User\User;
+
+class UserPolicy extends AbstractPolicy
+{
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $settings;
+
+    public function __construct(SettingsRepositoryInterface $settings)
+    {
+        $this->settings = $settings;
+    }
+    
+    /**
+     * @param User $actor
+     * @param User $user
+     * @return bool|null
+     */
+    public function editOwnNickname(User $actor, User $user)
```

Comment:
wouldn't it be better to have `editNickname` and in that we check if $actor === $user then we allow otherwise we check the normal `edit` ability ? rather than a `if ($isSelf) $actor->assert('editOwnNickname', $user)` in the listener